Sources/Sentry/SentryProfiler.mm

```diff
@@ -336,9 +380,9 @@ + (SentryEnvelopeItem *)createProfilingEnvelopeItemForTransaction:(SentryTransac
         return nil;
     }
 
-    NSMutableDictionary<NSString *, id> *payload = [_gCurrentProfiler->_profileData mutableCopy];
+    const auto payload = [NSMutableDictionary<NSString *, id> dictionary];
 
-    NSArray *samples = ((NSArray *)payload[@"profile"][@"samples"]).mutableCopy;
+    NSArray<SentrySample *> *samples = _gCurrentProfiler->_profileData[@"profile"][@"samples"];
```

Comment:
`_profileData` is a structure with nested dictionaries, and this copy operation only performed a shallow copy, and logic below wound up overwriting parts of it, and in tests of concurrent transactions, that wound up corrupting later serializations. Instead, I did away with any mutable copies and just create new structures, and manually assigned all the other separate substructures on lines 404-410 below.